pkg/netceptor/netceptor.go

```diff
@@ -607,23 +630,25 @@ func (s *Netceptor) getEphemeralService() string {
 // Prints the routing table.  Only used for debugging.
 // The caller must already hold at least a read lock on known connections and routing.
 func (s *Netceptor) printRoutingTable() {
-	logLevel, _ := logger.GetLogLevelByName("Debug")
-	if logger.GetLogLevel() >= logLevel {
+	debugLevel, _ := logger.GetLogLevelByName("Debug")
+	curLevel := logger.GetLogLevel()
+	if curLevel < debugLevel {
 		return
 	}
 	logger.Debug("Known Connections:\n")
 	for conn := range s.knownConnectionCosts {
-		logger.Debug("   %s: ", conn)
+		sb := &strings.Builder{}
+		_, _ = fmt.Fprintf(sb, "   %s: ", conn)
 		for peer := range s.knownConnectionCosts[conn] {
-			logger.Debug("%s(%.2f) ", peer, s.knownConnectionCosts[conn][peer])
+			_, _ = fmt.Fprintf(sb, "%s(%.2f) ", peer, s.knownConnectionCosts[conn][peer])
```

Comment:
im not sure i understand why this isnt a `logger.Info` output

Reply:
I think of this as a normal and routine operation that probably doesn't need to spam the logs, but I could be pretty easily talked into this being Info level.  Particularly since it only gets printed when something changes.

Reply:
hmm, yeah thinking about it more info is probably too high, but debug seems maybe appropriate. I was mostly just making sure this was intentional. I kinda expected we would want all output to go through the logger unless it cant because the logger isnt setup yet.

Reply:
In any case, we dont need to block on this, we can always change this later without much (any?) pain

Reply:
Oh, I see what you're saying.  All the output _is_ going through the logger.  The problem is that the logger adds a newline if one isn't already present, so you can't make multiple logger calls to build a single line the way I was originally doing with the printfs.  Instead, I'm constructing a whole line in the loop before sending it to the logger.  The `fmt.Fprintf` calls are writing _to the `string.Builder`_, not to the console.

Reply:
Ah! ok, yeah sorry i was confused